network-core/src/lib.rs

```diff
@@ -5,11 +5,37 @@
 #[macro_use]
 extern crate prost_derive;
 
+use chain_core::property::{Deserialize, Serialize};
+
 pub mod error;
 
 pub mod client;
 pub mod server;
 
+#[derive(Clone, Debug, PartialEq, PartialOrd)]
+pub struct NodeId([u8; 16]);
+
+pub enum Error {
+    InvalidSize(usize),
```

Comment:
This `Error` is now specific to node ID errors, but it is defined on the top level.
Furthermore, there is a mod `error` which is completely unrelated.

Maybe put this stuff along with `NodeId` and `Gossip` to a submodule, tentatively named `gossip`?